airflow/jobs/scheduler_job.py

```diff
@@ -336,7 +344,9 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session) -
                 query = query.filter(not_(TI.dag_id.in_(starved_dags)))
 
             if starved_tasks:
-                task_filter = tuple_in_condition((TaskInstance.dag_id, TaskInstance.task_id), starved_tasks)
+                task_filter = tuple_in_condition(
+                    (TaskInstance.dag_id, TaskInstance.run_id, TaskInstance.task_id), starved_tasks
```

Comment:
This method has a mixed use of `TaskInstance` and `TI`. While you are at it, could you unify the usage at least in this method and use `TI` everywhere.